src/components/Locations.js

```diff
@@ -0,0 +1,45 @@
+import {useState, useEffect} from "react"
+import axios from "axios"
+
+const Locations = () => {
+    const [locations, setLocations] = useState([])
+    const [showLocations, setShowLocations] = useState(false)
+
+    const fetchLocations = async () => {
+        try {
+            const res = await axios.get("https://ghibliapi.herokuapp.com/locations")
+            setLocations(res.data)
+        } catch (error) {
+            console.log(error);
+            setLocations([])
+        }
+    }
+
+    const handleClick = () => {
+        setShowLocations(prevShowLocations => !prevShowLocations)
+    }
+
+    useEffect(() => fetchLocations(), [])
```

Comment:
By using an implicit return here, when this component unmounts you'll also call fetchLocations. Thus, don't return here.